packages/react/src/internal/useResizeObserver.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useRef, useState, useLayoutEffect, useEffect, RefObject } from 'react';
+
+export const useResizeObserver = ({
+  ref,
+  onResize,
+}: {
+  ref: RefObject<HTMLElement>;
+  onResize?: (rect: DOMRectReadOnly) => void;
+}) => {
+  const [width, setWidth] = useState(-1);
+  const [height, setHeight] = useState(-1);
+  const entriesToHandle = useRef<ResizeObserverEntry[] | null>(null);
+  const cb = useRef(onResize);
+
+  useEffect(() => {
+    // ref for onResize removes it as dependency from useLayoutEffect
+    // This significantly reduces repeated calls if a function is redefined on every
+    // render
+    cb.current = onResize;
+  }, [onResize]);
+
+  useEffect(() => {
+    const getInitialSize = () => {
+      if (ref.current) {
+        const refComputedStyle = window.getComputedStyle(ref.current);
+
+        const initialWidth =
+          (ref.current?.offsetWidth || 0) -
```

Comment:
Suggestion to use a nullish coalescing operator here and in `initialHeight` to avoid `0 || 0`.

```suggestion
          (ref.current?.offsetWidth ?? 0) -
```